akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala

```diff
@@ -953,8 +965,8 @@ final class ClusterReceptionist(pubSubMediator: ActorRef, settings: ClusterRecep
         if (log.isDebugEnabled)
           log.debug("Client [{}] gets contactPoints [{}]", sender().path, contacts.contactPoints.mkString(","))
         sender() ! contacts
-        updateClientInteractions(sender())
       }
+      updateClientInteractions(sender())
```

Comment:
+1, that must have been an oversight

Reply:
Second thought about this. I don't think it is correct to call `updateClientInteractions` for `GetContacts` at all. That is only a request from the client to ask for "who should I talk to". It doesn't mean that it will establish the actual "connection" with this node. Adding it to the failure detect here must be wrong. That should only be done for heartbeat messages. AFAICS.

Reply:
I'm not sure what is the purpose of that feature. What you say above makes sense to me, but the documentation seems to match the current behavior, it doesn't mention an actual "connection" but just a "contact" ([source](https://doc.akka.io/docs/akka/current/cluster-client.html#events)):
> for learning what cluster clients contact a ``ClusterClientReceptionist``

Reply:
then the documentation should be updated to "what cluster clients are connected to a ClusterClientReceptionist"

The current implementation is not doing the same thing for the two cases above and I assume that `ClusterClientUnreachable` corresponds to "disconnected" and `ClusterClientUp` corresponds to "connected". 

I think this was implemented by @huntc . Any insights or objections?

Reply:
I'm sorry, I don't recall now.

Reply:
ok, then I think we can safely change the behavior as it's implemented in this PR now, and update the documentation